lib/plugins/event_dispatcher/tests.js

```diff
@@ -42,11 +42,15 @@ describe('lib/plugins/event_dispatcher', function() {
           },
```

Comment:
Can you add a test to `lib/optimizely/tests.js` where you pass in a custom event dispatcher that returns a promise and make sure that we are still able to dispatch correctly with that event dispatcher.